app/javascript/src/components/form/Derivatives.svelte

```diff
@@ -0,0 +1,84 @@
+<script>
+  import Tags from './Tags.svelte';
+  import { onMount } from 'svelte';
+
+  let showDerivative = false;
+  let maxCodes = 5;
+  export let currSources = [];
+
+  onMount(() => {
+    if (currSources.length) showDerivative = true;
+  });
+
+  async function handleAutoCompleteRequest(value) {
+    if (!value) return [];
+
+    const timeout = 8000;
+    const timeoutController = new AbortController();
+    const timeoutID = setTimeout(() => timeoutController.abort(), timeout);
+
+    const response = await fetch(`/code/${value}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      },
+      signal: timeoutController.signal
+    });
+    const text = await response.text();
+    clearTimeout(timeoutID);
+
+    if (response.ok) {
+      const json = JSON.parse(text);
+      return json.map(post => postToResult(post));
+    } else {
+      throw new Error(`${response.status} ${response.statusText}`);
+    }
+  }
+
+  function postToResult(post) {
+    return {
+      label: post.code,
+      html: `<strong>${post.code}</strong> - ${post.title} by ${post.user.username}`
+    };
+  }
+</script>
+
+<div class="form-group mt-1/4">
+  <div class="switch-checkbox">
+    <input
+      id="show_derivative"
+      class="switch-checkbox__input"
+      autocomplete="off"
+      type="checkbox"
+      bind:checked={showDerivative}
+    >
+    <label
+      class="switch-checkbox__label"
+      for="show_derivative"
+    >
+      This code is a fork/uses other codes
+    </label>
+  </div>
+
+  <div class="form-group mt-1/4" hidden={!showDerivative}>
```

Comment:
You could wrap this block in { #if showDerivative } ... { /if } instead of setting it to hidden. That way nothing gets render for those that never enable it in the first place.